ui/src/components/WorkersTable/index.jsx

```diff
@@ -110,11 +115,14 @@ export default class WorkersTable extends Component {
       Quarantined: node.quarantineUntil,
     };
 
-    return mapping[this.state.sortBy];
+    return mapping[query.sortBy];
   }
 
   render() {
-    const { sortBy, sortDirection } = this.state;
+    const query = parse(window.location.search.slice(1));
+    const { sortBy, sortDirection } = query.sortBy
+      ? query
+      : { sortBy: null, sortDirection: null };
     const {
       provisionerId,
       workerType,
```

Comment:
Good work on the refactor. Let's default to using `this.props.location` to have access to the location window rather than use window.location. For example:

```diff
diff --git a/ui/src/components/WorkersTable/index.jsx b/ui/src/components/WorkersTable/index.jsx
index 767b4b6c0..1e21182a0 100644
--- a/ui/src/components/WorkersTable/index.jsx
+++ b/ui/src/components/WorkersTable/index.jsx
@@ -89,8 +89,7 @@ export default class WorkersTable extends Component {
   );
 
   handleHeaderClick = sortByHeader => {
-    const { location } = this.props;
-    const query = parse(location.search.slice(1));
+    const query = parse(this.props.location.search.slice(1));
     const sortBy = sortByHeader;
     const toggled = query.sortDirection === 'desc' ? 'asc' : 'desc';
     const sortDirection = query.sortBy === sortBy ? toggled : 'desc';
@@ -103,7 +102,7 @@ export default class WorkersTable extends Component {
   };
 
   valueFromNode(node) {
-    const query = parse(window.location.search.slice(1));
+    const query = parse(this.props.location.search.slice(1));
     const mapping = {
       'Worker Group': node.workerGroup,
       'Worker ID': node.workerId,
@@ -119,7 +118,7 @@ export default class WorkersTable extends Component {
   }
 
   render() {
-    const query = parse(window.location.search.slice(1));
+    const query = parse(this.props.location.search.slice(1));
     const { sortBy, sortDirection } = query.sortBy
       ? query
       : { sortBy: null, sortDirection: null };
```

Let's replicate this logic to the other views as well.

Reply:
did the change